src/Nginx/Nginx.Autorest/test/utils.ps1

```diff
@@ -60,13 +61,80 @@ function setupEnv() {
+    $certKV = Add-AzKeyVaultCertificate -VaultName $env.keyvault -Name $env.nginxCert -CertificatePolicy $Policy
+
+    # getting the keyvault certificate secretid
+    $certVersion = "/" + (Get-AzKeyVaultcertificate -VaultName $env.keyvault -name $env.nginxCert).version 
+    $kvcertsecretid = (Get-AzKeyVaultcertificate -VaultName $env.keyvault -name $env.nginxCert).secretid.Replace(":443", "").Replace($certVersion, "")
+    
+    $env.kvcertsecretid = $kvcertsecretid
+    
     $envFile = 'env.json'
     if ($TestMode -eq 'live') {
         $envFile = 'localEnv.json'
     }
     set-content -Path (Join-Path $PSScriptRoot $envFile) -Value (ConvertTo-Json $env)
+
+    # create public ip
+    $ip = @{
+        Name              = $env.pubip
+        ResourceGroupName = $env.resourceGroup
+        Location          = $env.location
+        Sku               = "Standard"
+        AllocationMethod  = "Static"
+        IpAddressVersion  = "IPv4"
+        Zone              = 2
+    }
+    $publicIp = New-AzPublicIpAddress @ip -Force
+
+    # create virtual network
+    $vnet = @{
+        Name              = $env.vnet
+        ResourceGroupName = $env.resourceGroup
+        Location          = $env.location
+        AddressPrefix     = "10.0.0.0/16"
+    }
+    $virtualNetwork = New-AzVirtualNetwork @vnet -Force
+
+    # create subnet
+    $subnet = @{
+        Name           = $env.subnet
+        VirtualNetwork = $virtualNetwork
+        AddressPrefix  = "10.0.0.0/24"
+    }
+    $subnetConfig = Add-AzVirtualNetworkSubnetConfig @subnet
+    $virtualNetwork | Set-AzVirtualNetwork
+
+    # delegate the subnet to NGINX.NGINXPLUS/nginxDeployments
+    $vnet = Get-AzVirtualNetwork -Name $env.vnet -ResourceGroupName $env.resourceGroup
+    $subnet = Get-AzVirtualNetworkSubnetConfig -Name $env.subnet -VirtualNetwork $vnet
+    $subnet = Add-AzDelegation -Name "delegation" -ServiceName $env.delegation -Subnet $subnet
+    Set-AzVirtualNetwork -VirtualNetwork $vnet
+
+    # create the nginxaas resource
+    $publicIp = New-AzNginxPublicIPAddressObject -Id $publicIp.Id
+    $networkProfile = New-AzNginxNetworkProfileObject -FrontEndIPConfiguration @{PublicIPAddress = @($publicIp) } -NetworkInterfaceConfiguration @{SubnetId = $subnet.Id }
+    $nginxDeployment = New-AzNginxDeployment -Name $env.nginxDeployment1 -ResourceGroupName $env.resourceGroup -Location $env.location -NetworkProfile $networkProfile -SkuName standard_Monthly_gmz7xq9ge3py -IdentityType "SystemAssigned" 
+    $nginxDeployment.ProvisioningState | Should -Be "Succeeded"
+    $nginxDeployment.Name | Should -Be $env.nginxDeployment1
+
+    # assigning role
+    $keyVaultId = $keyVault.ResourceId
+    $roleDefinition = Get-AzRoleDefinition -Name "Key Vault Administrator"
+    $roleAssignment = New-AzRoleAssignment -ObjectId $nginxDeployment.IdentityPrincipalId  -RoleDefinitionId $roleDefinition.Id -Scope $keyVaultId
+    
 }
 function cleanupEnv() {
     # Clean resources you create for testing
```

Comment:
you might want to clean up resources created in setupEnv() here

Reply:
fixed, thank you